### ISSUE TITLE
unable to change viewportFraction on screen size change (web) 

### ISSUE THREAD

**Issue by akshaybhange** [serenader2014/flutter_carousel_slider#198]
When initially page builds it perfectly works, If I change screensize of my browser then entire widget builds but viewportfraction is not getting updated.  

```
CarouselSlider(
            carouselController: _carouselController,
            options: CarouselOptions(
              height: 300,
              enlargeCenterPage: false,
              pauseAutoPlayOnTouch: true,
              disableCenter: true,
              viewportFraction: _getViewportFraction(context),
              onPageChanged: (index, _) {
                setState(() {
                  _current = index;
                });
              },
              autoPlay: true,
            ),
            items: _getMagazineData()
                .map((e) => MagazineItem(magazine: e))
                .toList(),
          )
```

If i add `key: GlobalKey()` in `CarouselSlider` then my viewportFraction is updating but then slider & controller stops working (shows only same page again n again  and controller is not able to change to any page)

shows below given error 

```
Error: Assertion failed: file:///Users/setu-dev/flutter/packages/flutter/lib/src/widgets/page_view.dart:182:7
positions.isNotEmpty
"PageController.page cannot be accessed before a PageView is built with it."
    at Object.throw_ [as throw] (http://localhost:57293/dart_sdk.js:4322:11)
    at Object.assertFailed (http://localhost:57293/dart_sdk.js:4271:15)
    at page_view.PageController.new.get page [as page] (http://localhost:57293/packages/flutter/src/widgets/page_view.dart.lib.js:312:57)
    at http://localhost:57293/packages/carousel_slider/carousel_controller.dart.lib.js:941:58
    at http://localhost:57293/dart_sdk.js:23419:9
```

**Comment by akshaybhange** [comment]
seems like #174 is the solution to this.

**Comment by serenader2014** [comment]
#174 is merged, and published in v2.3.0. Please have a try.
